src/control/server/config/server.go

```diff
@@ -48,26 +49,27 @@ type deprecatedParams struct {
-	DisableVFIO       bool                      `yaml:"disable_vfio"`
-	DisableVMD        *bool                     `yaml:"disable_vmd"`
-	EnableHotplug     bool                      `yaml:"enable_hotplug"`
-	NrHugepages       int                       `yaml:"nr_hugepages"`        // total for all engines
-	SystemRamReserved int                       `yaml:"system_ram_reserved"` // total for all engines
-	DisableHugepages  bool                      `yaml:"disable_hugepages"`
-	ControlLogMask    common.ControlLogLevel    `yaml:"control_log_mask"`
-	ControlLogFile    string                    `yaml:"control_log_file,omitempty"`
-	ControlLogJSON    bool                      `yaml:"control_log_json,omitempty"`
-	HelperLogFile     string                    `yaml:"helper_log_file,omitempty"`
-	FWHelperLogFile   string                    `yaml:"firmware_helper_log_file,omitempty"`
-	FaultPath         string                    `yaml:"fault_path,omitempty"`
-	TelemetryPort     int                       `yaml:"telemetry_port,omitempty"`
-	CoreDumpFilter    uint8                     `yaml:"core_dump_filter,omitempty"`
-	ClientEnvVars     []string                  `yaml:"client_env_vars,omitempty"`
-	SupportConfig     SupportConfig             `yaml:"support_config,omitempty"`
+	ControlPort        int                       `yaml:"port"`
+	TransportConfig    *security.TransportConfig `yaml:"transport_config"`
+	Engines            []*engine.Config          `yaml:"engines"`
+	BdevExclude        []string                  `yaml:"bdev_exclude,omitempty"`
+	DisableVFIO        bool                      `yaml:"disable_vfio"`
+	DisableVMD         *bool                     `yaml:"disable_vmd"`
+	EnableHotplug      bool                      `yaml:"enable_hotplug"`
+	NrHugepages        int                       `yaml:"nr_hugepages"`        // total for all engines
+	SystemRamReserved  int                       `yaml:"system_ram_reserved"` // total for all engines
+	DisableHugepages   bool                      `yaml:"disable_hugepages"`
+	ControlLogMask     common.ControlLogLevel    `yaml:"control_log_mask"`
+	ControlLogFile     string                    `yaml:"control_log_file,omitempty"`
+	ControlLogJSON     bool                      `yaml:"control_log_json,omitempty"`
+	HelperLogFile      string                    `yaml:"helper_log_file,omitempty"`
+	FWHelperLogFile    string                    `yaml:"firmware_helper_log_file,omitempty"`
+	FaultPath          string                    `yaml:"fault_path,omitempty"`
+	TelemetryPort      int                       `yaml:"telemetry_port,omitempty"`
+	CoreDumpFilter     uint8                     `yaml:"core_dump_filter,omitempty"`
+	ClientEnvVars      []string                  `yaml:"client_env_vars,omitempty"`
+	SupportConfig      SupportConfig             `yaml:"support_config,omitempty"`
+	ClientFirewallMode *bool                     `yaml:"client_behind_firewall,omitempty"`
```

Comment:
This kind of inconsistency will cause problems for people trying to follow the code. The YAML name should match the field name unless there is a specific reason not to and IMO that should follow the flag used in the command.

Reply:
`client_behind_firewall` is user facing, `ClientFirewallMode` is programmer facing. Anyways, it will lead to large changes if we need to modify it, and we have landed the PR internally already, so let's keep it as-is

Reply:
I can live with it at other levels where we use the "client firewall mode" terminology, but I do think at the config level it is good to have the parameter match the name.